pytorch_lightning/metrics/self_supervised.py

```diff
@@ -0,0 +1,75 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Optional, Sequence, Tuple
+
+import torch
+
+from pytorch_lightning.metrics.functional.self_supervised import embedding_similarity
+from pytorch_lightning.metrics.metric import TensorMetric
+
+
+class EmbeddingSimilarity(TensorMetric):
+    """
+
+
+    """
+
+    def __init__(self,
+                 similarity: str = 'cosine',
+                 zero_diagonal: bool = True,
+                 reduction: str = 'mean',
+                 reduce_group: Any = None):
+        """ Computes similarity between embeddings
+
+            Example:
+
+                >>> embeddings = torch.tensor([[1., 2., 3., 4.], [1., 2., 3., 4.], [4., 5., 6., 7.]])
+                >>> embedding_similarity(embeddings)
+                tensor([[0.0000, 1.0000, 0.9759],
+                        [1.0000, 0.0000, 0.9759],
+                        [0.9759, 0.9759, 0.0000]])
+
+            Args:
+                similarity: 'dot' or 'cosine'
+                reduction: 'none', 'sum', 'mean' (all along dim -1)
+                zero_diagonal: if True, the diagonals are set to zero
+                reduce_group: the process group to reduce metric results from DDP
```

Comment:
there should be one less level of indentation to make it consistent with other docs, and otherwise the generated html has very strange formatting.

Reply:
should be fixed now